src/entities/person/person.ts

```diff
@@ -984,6 +1002,39 @@ export class Person extends UniverseEntity<PersonPayload, PersonRawPayload> {
       throw this.handleError(new PossibleDuplicatesFetchRemoteError(undefined, { error: err }))
     }
   }
+
+  /**
+   * Unified sugar method to handle all the different aspects of a person messageSubscriptionInstance
+   *
+   * @param messageSubscriptionId
+   * @param action - 'optin' | 'optout' | 'withdraw'
+   * @param body  - optional context
+   * @param options - optional query parameters
+   */
+  public async handleMessageSubscription (messageSubscriptionId: string, action: IMessageSubscriptionActionType, body?: IHandleMessageSubscriptionBody, options?: EntityFetchOptions): Promise<MessageSubscriptionInstanceRawPayload | MessageSubscriptionInstance> {
+    if (!(this.id && messageSubscriptionId && action)) throw new TypeError('handleMessageSubscription requires personId, messageSubscriptionId and action (optin, optout, withdraw) to be set.')
+
+    try {
+      const opts = {
+        method: 'POST',
+        url: `${this.universe.universeBase}/api/v0/people/${this.id}/message_subscriptions/${messageSubscriptionId}/${action}${options?.query ? qs.stringify(options.query, { addQueryPrefix: true }) : ''}`,
+        data: body,
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8'
+        },
+        responseType: 'json'
+      }
+      const res = await this.http?.getClient()(opts)
+      const resource = res.data.data?.[0] as MessageSubscriptionInstanceRawPayload
```

Comment:
Could possibly be also undefined right? In that case the type casting should probably reflect that as well